### PR TITLE
Fx2 / delay lp aliasing / Diffuser click on change

### PR DESCRIPTION
**Pull request by pvig**

### CODE REVIEW

File: firmware/Src/synth/Timbre.h [hunk 1]
```diff
@@ -228,6 +228,10 @@ class Timbre {
     float low6 = 0, band6 = 0;
     float low7 = 0, band7 = 0;
     float low8 = 0, band8 = 0;
+    float low9 = 0, band9 = 0;
```
Comment (Ixox) [inline firmware/Src/synth/Timbre.h:231]: Is it possible to use the same variables ? 
We'll soon be out of memory for the fast access RAM.

File: firmware/Src/synth/Timbre.cpp [hunk 2]
```diff
@@ -1960,27 +1963,37 @@ void Timbre::fxAfterBlock() {
...
+                band5 += f2 * (low4 - low5 - band5);
+                low6  += f2 * band6;
+                band6 += f2 * (low5 - low6 - band6);
 
                 // lp R
-                low4  += f2 * band4;
-                band4 += f2 * ((delayOut2) - low4 - band4);
-                low6  += f2 * band6;
-                band6 += f2 * ((low4) - low6 - band6);
+                low9  += f2 * band9;
```
Comment (Ixox) [inline firmware/Src/synth/Timbre.cpp:1979]: I must miss something.
How can we have only "+=" on the low* and band* variable ?
We never reset their value ? (except when we instanciate the object).

File: firmware/Src/synth/Timbre.h (hunk 1 above)
Comment (patvig) [reply firmware/Src/synth/Timbre.h:231]: yes, i'll do some refactoring tonight to clean up varaibles allocation.

File: firmware/Src/synth/Timbre.cpp (hunk 2 above)
Comment (patvig) [reply firmware/Src/synth/Timbre.cpp:1979]: That's the way these lowpass filters works : input is fed into the band* variable then read with the low* var.

Comment (pvig) [reply firmware/Src/synth/Timbre.cpp:1979]: i reckon i do not fully understand what's happening within these filters equations, but it works !

**Review by pvig — COMMENTED** [review]
Variables cleaned

File: firmware/Src/synth/Timbre.cpp [hunk 3]
```diff
@@ -917,13 +887,22 @@ void Timbre::fxAfterBlock() {
             feedback = clamp(this->params_.effect2.param2 + matrixFilterParam2, -1, 1) * 0.4f;
             float feedbackInc = (feedback - currentFeedback) * INV_BLOCK_SIZE;
 
+            float delayIn;
```
Comment (Ixox) [inline firmware/Src/synth/Timbre.cpp:890]: Not used ?

File: firmware/Src/synth/Timbre.cpp [hunk 4]
```diff
@@ -1660,12 +1655,12 @@ void Timbre::fxAfterBlock() {
...
-                    hp_in3_y0    = _in3_a0 * hp_in3_x0 + _in3_a1 * hp_in3_x1 + _in3_b1 * hp_in3_y1;
-                    hp_in3_y1    = hp_in3_y0;
-                    hp_in3_x1    = hp_in3_x0;
+                    hb5_x1    = hp_in2_y0 + monoIn;
+                    hb5_y1    = _in3_a0 * hb5_x1 + _in3_a1 * hb5_x2 + _in3_b1 * hb5_y2;
+                    hb5_y2    = hb5_y1;
+                    hb5_x2    = hb5_x1;
 
-                    float delayIn = hp_in3_y0;
+                    float delayIn = hb5_y1;
```
Comment (Ixox) [inline firmware/Src/synth/Timbre.cpp:1663]: hb5_y1 can be used in the line bellow.

File: firmware/Src/synth/Timbre.h [hunk 5]
```diff
@@ -204,34 +201,24 @@ class Timbre {
...
-    float _in_lp2_a, _in_lp2_b;
-    float lpF, lpF2;
      
     float low1 = 0, band1 = 0;
     float low2 = 0, band2 = 0;
     float low3 = 0, band3 = 0;
     float low4 = 0, band4 = 0;
     float low5 = 0, band5 = 0;
     float low6 = 0, band6 = 0;
-    float low7 = 0, band7 = 0;
```
Comment (Ixox) [inline firmware/Src/synth/Timbre.h:229]: WOW....  quel netoyage !!! 👍🏼

**Review by pvig — COMMENTED** [review]
more cleaning

**Review by pvig — COMMENTED** [review]
diffuser 5 bug fix
il était la depuis un moment celui la..

**Review by pvig — COMMENTED** [review]
a bit of tweaking for the Diff fx

**Comment by pvig** [comment]
Seems ok ?

**Comment by Ixox** [comment]
I've just merged the other changes in master.
I'll take take a look at this one.

Le sam. 27 mai 2023 à 10:49, patrice Vigouroux ***@***.***> a
écrit :

**Comment by Ixox** [comment]
Merged

**Pull request closed without merging**